src/gui/AutomationPatternView.cpp

```diff
@@ -340,7 +340,7 @@ void AutomationPatternView::paintEvent( QPaintEvent * )
 	const int lineSize = 3;
 	p.setPen( c.darker( 300 ) );
 
-	for( tact_t t = 1; t < m_pat->timeMapLength().getTact(); ++t )
+	for( tact_t t = 1; t < width() - TCO_BORDER_WIDTH; ++t )
 	{
 		const int tx = x_base + static_cast<int>( ppt * t ) - 2;
 		if( tx < ( width() - TCO_BORDER_WIDTH * 2 ) )
```

Comment:
Perhaps it's possible to remove this if( ... ) since the tests now look similar?